### PR TITLE
upgrade for run COPY in psql

### PR DESCRIPTION
**Pull request by RomanTkachuk**

### CODE REVIEW

File: lib/DBIx/Migration.pm [hunk 1]
```diff
@@ -272,14 +272,14 @@ sub _connect {
     return;
 }
 
-sub psql {
-    my ($self, @filenames) = @_;
+sub psql_full {
+    my ($self, $cmd_before, $cmd_after, @filenames) = @_;
```
Comment (torsten-deriv) [inline lib/DBIx/Migration.pm]: you could make this backwards compatible this way:

``` perl
sub psql {
    my ($self, @filenames) = @_;
    my %options;
    %options = @{shift @filenames} if ref $filenames[0];

    ...

    print $psql_in "$options{before};\n" or die "Cannot write to psql: $!\n" if $options{before};

    ....
}
```

File: lib/DBIx/Migration.pm [hunk 2]
```diff
@@ -272,14 +272,14 @@ sub _connect {
...
 
     my ($fh, $fn) = tempfile undef, UNLINK => 1;
     $fh->autoflush(1);
 
     $self->_connect unless $self->{_dbh}->{Active};
-    my ($dbname, $dbuser, $dbhost, $dbport, $dbpass) = @{$self->{_dbh}}{qw/pg_db pg_user pg_host pg_port pg_pass/};
+    my ($dbname, $dbuser, $dbhost, $dbport, $dbpass) = @{$self->{_dbh}}{qw/pg_db pg_ user pg_host pg_port pg_pass/};
```
Comment (torsten-deriv) [inline lib/DBIx/Migration.pm]: that's a bug

File: lib/DBIx/Migration.pm [hunk 3]
```diff
@@ -314,6 +320,12 @@ sub psql {
     die "psql returns unexpected code $rc\n";
 }
 
+sub psql {
```
Comment (torsten-deriv) [inline lib/DBIx/Migration.pm]: Oh I see. You chose this way for compatibility.

Also possible. I don't mind

File: lib/DBIx/Migration.pm [hunk 4]
```diff
@@ -296,9 +298,15 @@ sub psql {
     }
 
     print $psql_in "SET client_min_messages TO warning;\n" or die "Cannot write to psql: $!\n";
+    if ($options->before) {
```
Comment (torsten-deriv) [inline lib/DBIx/Migration.pm]: `$options{before}`

File: lib/DBIx/Migration.pm [hunk 5]
```diff
@@ -296,9 +298,15 @@ sub psql {
...
+    	 print $psql_in "$options{before};\n" or die "Cannot write to psql: $!\n";
+    }
     for my $name (@filenames) {
         print $psql_in "\\i $name\n" or die "Cannot write to psql: $!\n";
     }
+    if ($options->after) {
```
Comment (torsten-deriv) [inline lib/DBIx/Migration.pm]: same here

**Comment by shuwnyuantee** [comment]
What is this `before` , `after` options used for?

**Pull request merged**